src/pages/staking/Staking.vue

```diff
@@ -193,6 +199,24 @@ export default {
                 }
             },
         },
+        $route: {
```

Comment:
this is identical to the implementation in `Transaction.vue`, refactor if practical.